### PR TITLE
SDIT-1421 Added Incident requirements and questions

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/IncidentService.kt [hunk 1]
```diff
@@ -51,13 +54,25 @@ class IncidentService(
...
+    prisonId = location.id,
+  )
+private fun IncidentQuestion.toQuestion() =
+  Question(question = question.question)
+
 private fun Incident.staffParties(): List<IncidentParty> =
   this.parties.filter { it.isStaffParty() }
 
-private fun uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Staff.toStaff() =
+private fun JPAStaff.toStaff() =
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/IncidentService.kt:75]: suggestion from previous PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/IncidentQuestion.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.IncidentQuestion
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.IncidentQuestionId
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.QuestionnaireQuestion
+
+@DslMarker
+annotation class IncidentQuestionDslMarker
+
+@NomisDataDslMarker
+interface IncidentQuestionDsl {
+  /*
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/IncidentQuestion.kt:14]: incident response work temporarily commented out until fully working

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/IncidentResourceIntTest.kt [hunk 3]
```diff
@@ -74,6 +84,15 @@ class IncidentResourceIntTest : IntegrationTestBase() {
       ) {
         incidentParty(staff = partyStaff1)
         incidentParty(offenderBooking = offenderParty.latestBooking(), outcome = "POR")
+        requirement("Update the name", recordingStaff = requirementRecordingStaff, prisonId = "MDI")
+        requirement("Ensure all details are added", recordingStaff = requirementRecordingStaff, prisonId = "MDI")
+        question(question = question1)
+        question(question = question2) {
+          //  response(
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/IncidentResourceIntTest.kt:91]: temporarily commented out until incident response work finished

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**